zim.cc

```diff
@@ -42,6 +42,8 @@
 #include <string>
 #include <set>
 #include <map>
+#include <algorithm>
+#include <functional>
```

Comment:
`<functional>` should be included only in the files where you use `std::greater`.